Add "Jump emitter forward in time on creation" property

tristanbob opened the pull request:
Following these instructions from @4ian: https://github.com/4ian/GDevelop/pull/4985#issuecomment-1443120194

tristanbob commented:
The property seems to be working properly (the editor looks good and it saves changes to the value).

However, the emitter is not jumping forward in time.  

Did I call the update() function correctly?

Did I put it in the right location of the constructor?

4ian commented on Extensions/ParticleSystem/particleemitterobject.ts:
```diff
@@ -158,8 +160,13 @@ namespace gdjs {
       this._textureDirty = this.texture !== '';
 
+      // Jump emitter forward in time
+      this._renderer.update(this.jumpForwardInTimeOnCreation);
```
Try to add a console.log here to see if `this.jumpForwardInTimeOnCreation` has a proper value?

My hypothesis is that it's maybe too early to ask the renderer to update?

tristanbob commented:
I got this working! 

I moved the function call from the constructor to update() and I added a boolean variable to function as "TriggerOnce".  

I have a few questions:

- Where is the best place to insert new functions/properties, etc? I have been trying to put them at the bottom of the namespace, except when something seemed to be at the bottom for a reason.
- Why did we need to define the properties twice in "ParticleSystem\particleemitterobject.ts" (once in `ParticleEmitterObjectDataType` and once in `ParticleEmitterObject`)?
- Was it correct to define my boolean `_jumpForwardInTimeCompleted` twice also?

Thanks for the help!

tristanbob commented:
Does anyone have time to answer my questions on this PR?  (see my previous comment)

4ian commented:
> Where is the best place to insert new functions/properties, etc? I have been trying to put them at the bottom of the namespace, except when something seemed to be at the bottom for a reason.

Generally, adding after the rest is ok!

> Why did we need to define the properties twice in "ParticleSystem\particleemitterobject.ts" (once in ParticleEmitterObjectDataType and once in ParticleEmitterObject)?

`ParticleEmitterObjectDataType` describes what you're expecting to read from the project. 
`ParticleEmitterObject` is what lives in memory during the game. 
These are often similar, but not necessarily (typically, the boolean to know if something was done has no sense in the `ParticleEmitterObjectDataType`)


> Was it correct to define my boolean _jumpForwardInTimeCompleted twice also?

See my comment ;)

4ian commented:
Apart from one comment, I think this will be ready to merge!

tristanbob commented on Extensions/ParticleSystem/particleemitterobject.ts:
```diff
@@ -52,6 +52,8 @@ namespace gdjs {
     /** Destroy the object when there is no particles? */
     destroyWhenNoParticles: boolean;
+    jumpForwardInTimeOnCreation: number;
+    _jumpForwardInTimeCompleted: boolean;
```
```suggestion
```

tristanbob commented:
> > Was it correct to define my boolean _jumpForwardInTimeCompleted twice also?
> 
> See my comment ;)

Thanks, I have removed the line you suggested.

(I accidentally made the change as a new comment instead of directly resolving your comment, but now I know how to do it on the next review).

BTW - I did a quick build with this commit and it worked as expected.
